pkg/types/ccipocr3/config.go

```diff
@@ -78,3 +64,148 @@ type ObserverInfo struct {
 	// Reads define the chains that the current node can read from.
 	Reads []ChainSelector `json:"reads"`
 }
+
+type P2PToSupportedChains map[P2PID]SupportedChains
+
+func (cm *P2PToSupportedChains) UnmarshalJSON(data []byte) error {
+	temp := make(map[string]SupportedChains)
+	if err := json.Unmarshal(data, &temp); err != nil {
+		return err
+	}
+	//if temp == nil {
+	//	*cm = nil
+	//	return nil
+	//}
+	*cm = make(P2PToSupportedChains)
+	for k, v := range temp {
+		var key P2PID
+		decoded, err := hex.DecodeString(k)
+		if err != nil {
+			return err
+		}
+		copy(key[:], decoded)
+		(*cm)[key] = v
+	}
+	return nil
+}
+
+func (cm P2PToSupportedChains) MarshalJSON() ([]byte, error) {
+	temp := make(map[string]SupportedChains)
+	for k, v := range cm {
+		temp[hex.EncodeToString(k[:])] = v
+	}
+	return json.Marshal(temp)
+}
+
+type ConsensusObservation struct {
```

Comment:
This name is too general, I suggest calling it something more specific since we come to consensus on a lot of things